crates/ruff_python_parser/src/parser/expression.rs

```diff
@@ -2251,9 +2256,15 @@ impl<'src> Parser<'src> {
             return self.parse_yield_from_expression(start);
         }
 
-        let value = self
-            .at_expr()
-            .then(|| Box::new(self.parse_star_expression_list().expr));
+        let value = self.at_expr().then(|| {
+            Box::new(
+                self.parse_expression_list(
+                    ExpressionContext::default()
+                        .with_starred_expression_allowed(StarredExpressionPrecedence::BitwiseOr),
+                )
```

Comment:
I'm not sure if we should inherit the context here, to avoid unsetting the `INCLUDE_IN`


```suggestion
                    context
                        .with_starred_expression_allowed(StarredExpressionPrecedence::BitwiseOr),
                )
```

Reply:
I'm assuming this is specific to the yield expression for which I commented on the PR. If so, I'll create a follow-up to pass in the context to yield expressions in a later PR. I think it makes sense to avoid consuming the `in` keyword in a yield expression.